### ISSUE TITLE
Bad suggestion for "etant"

### ISSUE THREAD

**Issue by danielnaber** [languagetool-org/languagetool#1633]
Should be "étant" but isn't for some reason.

**Comment by dpelle** [comment]
The ticket is rather short.  I assume we are talking about spelling suggestions and not a about grammar rule.  "etant" is not a valid word in French.  The spelling corrections proposed by LT are
not so good indeed.  Using the web demo, it proposes:
* tant
* état
* jetant
* étang
* Ethan

A spelling mistake with missing diacritics should not be penalized much. Certainly "étant" should be the top suggestion.

Actually, I don't know how to upgrade the French dictionary anymore.  It was using Hunspell earlier (which most certainly had batter suggestion but was slow), and now it uses a binary FSA dictionary.  It would be good to have a script that completely automates the creation of the French dictionary (just like any other binary dictionary) otherwise it's not well maintainable.  I see that the binary FSA spelling dictionary was introduced in this commit:
```
$ git log fr_FR.dictcommit 9ef84c6e38ac3c03d89ed1db53ed37213c4e1003
Author: Daniel Naber <naber@danielnaber.de>
Date:   Thu Dec 21 15:54:25 2017 +0100

    [fr] offer suggestions for spelling errors
```
There is a `fr_FR.README` text file but it's vague about how to re-create the dictionary.  A script would be more precise:

```
$ cat languagetool/languagetool-language-modules/fr/src/main/resources/org/languagetool/resource/fr/hunspell/fr_FR.README

This Morfologik dictionary (fr_FR.dict, fr_FR.info) is based on a word list exported by
https://github.com/Fanaen/Hunspell2WordList. The following script has then been used to generate the
binary Morfologik dictionary: https://gist.github.com/danielnaber/b194b1a53e5a76373763e206c0ccbf02

The license is that of the original data, namely MPL 2.0 (see ../README_lexique.txt) 

Also see https://github.com/languagetool-org/languagetool/issues/850
```

**Comment by danielnaber** [comment]
Indeed, when I export the binary (Morfologik) dictionary with `java -cp languagetool.jar org.languagetool.tools.DictionaryExporter -i org/languagetool/resource/fr/hunspell/fr_FR.dict -info org/languagetool/resource/fr/hunspell/fr_FR.info -o /tmp/out.txt`,  "étant" is simple not in that list. That seems to be the problem. I'll spend some more time trying to debug/fix this.

**Comment by danielnaber** [comment]
The result of Hunspell2WordList, which tries to export the hunspell dict into a flat word list, just doesn't contain "étant". Maybe because it's not in the hunspell *.dic, but in the *.aff:

`SFX zE être étant/n'q'l'm't's' être po:ppre`

The same seems true for other words, e.g. foutant and mettant. Using `unmunch`, these words are exportet, but unmuch has different issues (see #850).

**Comment by danielnaber** [comment]
https://github.com/Fanaen/Hunspell2WordList/issues/2 has been fixed, so after 4.6, we can update the dictionary (also using the latest version of Dicollecte, 6.4.1)

**Comment by danielnaber** [comment]
Fix should be online tomorrow.
